lib/Settings/Admin.php

```diff
@@ -0,0 +1,175 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\Files_External_BeeSwarm\Settings;
+
+use OCA\Files_External\Service\BackendService;
+use OCA\Files_External\Service\GlobalStoragesService;
+use OCP\AppFramework\Http\TemplateResponse;
+use OCP\IConfig;
+use OCP\IL10N;
+use OCP\Settings\ISettings;
+
+class Admin implements ISettings {
+
+	/** @var string */
+	protected $appName;
+
+	/** @var IConfig */
+	private $config;
+	/** @var IL10N */
+	private $l;
+
+	/** @var GlobalStoragesService */
+	private $globalStoragesService;
+
+	/** @var BackendService */
+	private $backendService;
+
+	public function __construct($appName,
+								IConfig $config,
+								IL10N $l,
+								GlobalStoragesService $globalStoragesService,
+								BackendService $backendService) {
+		$this->appName = $appName;
+		$this->config = $config;
+		$this->l = $l;
+		$this->globalStoragesService = $globalStoragesService;
+		$this->backendService = $backendService;
+	}
+
+	/**
+	 * @return TemplateResponse
+	 */
+	public function getForm(): TemplateResponse {
+		// Get Mount settings
+		$mounts_json = $this->config->getAppValue($this->appName,"storageconfig","");	//default
+		$mounts = json_decode($mounts_json,  true);
+
+		// Get all valid Beeswarm storages
+		$storageBackends = array_filter($this->globalStoragesService->getStorages(), function ($storageBackend) use ($_) {
+			return $storageBackend->getBackend()->getStorageClass() == '\OCA\Files_External_BeeSwarm\Storage\BeeSwarm';
+		});
+
+		// remove configurations that are no longer in Storages
+		foreach ($mounts as $key => $mount) {
+			$eachmountId = $mount['mount_id'];
+			if (!isset($storageBackends[$eachmountId])) {
+				// remove by key index
+				unset($mounts[$key]);
+			}
+		}
+
+		$mountIds = array_column($mounts, 'mount_id');
+		foreach ($storageBackends as $backend) {
+			$backendId = $backend->getId();
+			$backendName = $backend->getMountPoint();
+			$urlOptions = $backend->getBackendOptions();
+
+			// Is it configured?
+			$key = array_search($backendId, $mountIds);
+			if (!empty($key) || $key === 0) {
+				$mounts[$key]['mount_name'] = $backendName;
+				$batchId = isset($mounts[$key]['batchid']) ? $mounts[$key]['batchid'] : "";
+				$mounts[$key]['batchbalance'] = $this->getBatchBalance($batchId, $urlOptions);
+				$mounts[$key]['chequebalance'] = $this->getChequebookBalance($urlOptions);
+			}
+			else {
+				// not yet configured, so add to array. Default to encryption.
+				$newMounts[] = ['mount_id' => $backendId, 'mount_name' => $backendName, 'encrypt'=> '1'];
+			}
+		}
+		if ($newMounts) {
+			$mounts = array_merge($mounts, $newMounts);
+		}
+
+		$parameters = [
+			'visibilityType' => BackendService::VISIBILITY_ADMIN,
```

Comment:
You're instantiating the backendservice in the constructor but the only call to it is this, so the instantiated version isn't actually being called. If we only need a constant from the class I think we can remove it from the constructor.

Reply:
Done